### PR TITLE
make the kdump initrd before the FS becomes read only

### PR DESCRIPTION
**Pull request by mvidner**
https://trello.com/c/GeRaiaKi
https://bugzilla.suse.com/show_bug.cgi?id=1022496

The actual fix is in commit 1d9ef41156673ad77c429cc0fff4bf3d0ef4db95, the rest is refactoring.

### CODE REVIEW

File: src/lib/kdump/clients/finish.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+      def steps
+        3
+      end
+
+      def write
+        progress_orig = Progress.set(false)
+        # propose settings for kdump
+        # if autoyast doesn't include settings for yast2-kdump
+        Kdump.Propose if !Kdump.import_called && (Mode.autoinst || Mode.autoupgrade)
```
Comment (lslezak) [inline src/lib/kdump/clients/finish.rb]: Use `Mode.auto` here, it's equivalent to `Mode.autoinst || Mode.autoupgrade`.

Comment (mvidner) [inline src/lib/kdump/clients/finish.rb]: Yes, but that's new API, not in SP2.

Comment (teclator) [inline src/lib/kdump/clients/finish.rb]: new API?

Comment (mvidner) [inline src/lib/kdump/clients/finish.rb]: Ah, I was wrong, it is in SP2: https://github.com/yast/yast-yast2/commit/3a6987c12efb12fe6a8e5605b897f1d9b04ef0d8

**Pull request merged**